Wikipedia/Code/UIViewController+WMFArticlePresentation.m

```diff
@@ -42,10 +42,12 @@ - (void)wmf_pushArticleViewController:(WMFArticleViewController *)viewController
     }
     [[PiwikTracker wmf_configuredInstance] wmf_logView:viewController];
 
-    dispatchOnMainQueueAfterDelayInSeconds(0.5, ^{
-        MWKHistoryList *historyList = viewController.dataStore.userDataStore.historyList;
-        [historyList addPageToHistoryWithURL:viewController.articleURL];
-    });
+    if(viewController.isAddingArticleToHistoryListEnabled){
+        dispatchOnMainQueueAfterDelayInSeconds(0.5, ^{
```

Comment:
@montehurd i forgot to say, you need to remove the delay before testing… but i figure you know anyways